clarity/src/vm/analysis/type_checker/contexts.rs

```diff
@@ -86,6 +86,25 @@ impl TypeMap {
         }
     }
 
+    /// Like set_type but forcing a change if already set
+    pub fn override_type(&mut self, expr: &SymbolicExpression, type_sig: TypeSignature) {
```

Comment:
Shouldn’t this method checks that the `expr` already exists? Otherwise, what are we overriding?